lambdas/thumbnail/index.py

```diff
@@ -218,58 +232,75 @@ def lambda_handler(request):
     # Parse request info
     url = request.args['url']
     size = SIZE_PARAMETER_MAP[request.args['size']]
+    input_ = request.args.get('input', 'image')
     output = request.args.get('output', 'json')
+    page = request.args.get('page', 1)
 
     # Handle request
     resp = requests.get(url)
     if resp.ok:
-        # Get the original reader / format
-        # If the original reader isn't in the supported formats map, use PNG as default presentation format
         try:
             thumbnail_format = SUPPORTED_BROWSER_FORMATS.get(imageio.get_reader(resp.content), "PNG")
-        # In the case imageio can't read the image, default to PNG
-        # Usually an OME-TIFF / CZI / some other bio-format
         except ValueError:
-            thumbnail_format = "PNG"
-
-        # Read image data
-        img = AICSImage(resp.content)
-        orig_size = list(img.reader.data.shape)
-
-        # Generate a formatted ndarray using the image data
-        # Makes some assumptions for n-dim data
-        img = format_aicsimage_to_prepped(img)
-
-        # Send to Image object for thumbnail generation and saving to bytes
-        img = Image.fromarray(img)
-
-        # Generate thumbnail
-        img.thumbnail(size)
-        thumbnail_size = img.size
-
-        # Store the bytes
-        thumbnail_bytes = BytesIO()
-        img.save(thumbnail_bytes, thumbnail_format)
-
-        # Get bytes data
-        data = thumbnail_bytes.getvalue()
-
-        # Create metadata object
-        info = {
-            'original_size': orig_size,
-            'thumbnail_format': thumbnail_format,
-            'thumbnail_size': thumbnail_size,
-        }
+            thumbnail_format = "JPEG" if input_ == "pdf" else "PNG"
+        if input_ == "pdf":
+            try:
+                pages = convert_from_bytes(
+                    resp.content,
+                    first_page=page,
+                    last_page=page,
+                    # respect width but not necessarily height (to preserve aspect ratio)
+                    size=(size[0], None)
+                )
+            except (
+                    PDFInfoNotInstalledError,
+                    PDFPageCountError,
+                    PDFSyntaxError,
+                    PopplerNotInstalledError
+            ) as exc:
+                return make_json_response(500, {'error': str(exc)})
```

Comment:
I have been thinking about failed thumbnails for a bit. It may be nice to have a default image when any thumbnail render fails. So instead of just showing the text of the filename it shows some static image of "Could not render preview" or similar.

Reply:
Good idea. We could do that in the catalog. CC @nl0.

Reply:
backlogged